src/core/MOM_stoch_eos.F90

```diff
@@ -184,10 +184,9 @@ subroutine MOM_calc_varT(G,Gv,h,tv,stoch_eos_CS)
   enddo
   ! if stochastic, perturb
   if (stoch_eos_CS%use_stoch_eos) then
-     call pass_var(stoch_eos_CS%pattern, G%Domain)
```

Comment:
Do we need this halo update? I don't think any other parts of the code use `stoch_eos_CS%pattern`, they just use `tv%varT`. The loop below _used_ to access `stoch_eos_CS%pattern` outside the computational domain, which would have required a halo update to get the right values. The lines immediately below this one are changed to only use `stoch_eos_CS%pattern` to update `tv%varT` on the computational domain. So it seems to me that we could drop this halo update, and the one new line in MOM.F90 will make sure that the values are correct anytime a processor accesses `tv%varT` outside the computational domain.